addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2382,7 +2384,7 @@ var SnippetsMenu = Widget.extend({
                         model: 'ir.ui.view',
                         method: 'delete_snippet',
                         kwargs: {
-                            'view_id': parseInt(ev.currentTarget.dataset.snippetId),
+                            'view_id': snippetId,
```

Comment:
The issue was that this line was reading the event's currentTarget asynchronously, so if any other delegate handler was registered on this widget and triggered on the same click, it would alter the currentTarget property before we had read it. This commit fixes that by reading the information that we need (the snippet id) synchronously at the start of the handler